typescript/docs/workflows.md

```diff
@@ -349,53 +349,65 @@ _Source: [examples/workflows/contentCreator.ts](/typescript/examples/workflows/c
 
 ```ts
 import "dotenv/config";
-import { UnconstrainedMemory } from "beeai-framework/memory/unconstrainedMemory";
 import { createConsoleReader } from "examples/helpers/io.js";
 import { OpenMeteoTool } from "beeai-framework/tools/weather/openMeteo";
 import { WikipediaTool } from "beeai-framework/tools/search/wikipedia";
-import { AgentWorkflow } from "beeai-framework/workflows/agent";
-import { UserMessage } from "beeai-framework/backend/message";
+import { AgentWorkflow, AgentWorkflowInput } from "beeai-framework/workflows/agent";
 import { WatsonxChatModel } from "beeai-framework/adapters/watsonx/backend/chat";
 
-const workflow = new AgentWorkflow();
+const workflow = new AgentWorkflow("Smart assistant");
 const llm = new WatsonxChatModel("meta-llama/llama-3-3-70b-instruct");
 
 workflow.addAgent({
-  name: "WeatherForecaster",
-  instructions: "You are a weather assistant. Respond only if you can provide a useful answer.",
-  tools: [new OpenMeteoTool()],
+  name: "Researcher",
+  role: "A diligent researcher.",
+  instructions: "You look up and provide information about a specific topic.",
+  tools: [new WikipediaTool()],
   llm,
-  execution: { maxIterations: 3 },
 });
 workflow.addAgent({
-  name: "Researcher",
-  instructions: "You are a researcher assistant. Respond only if you can provide a useful answer.",
-  tools: [new WikipediaTool()],
+  name: "WeatherForecaster",
+  role: "A weather reporter.",
+  instructions: "You provide detailed weather reports.",
+  tools: [new OpenMeteoTool()],
   llm,
+  execution: { maxIterations: 3 },
 });
 workflow.addAgent({
-  name: "Solver",
-  instructions:
-    "Your task is to provide the most useful final answer based on the assistants' responses which all are relevant. Ignore those where assistant do not know.",
+  name: "DataSynthesizer",
+  role: "A meticulous and creative data synthesizer",
+  instructions: "You can combine disparate information into a final coherent summary.",
   llm,
 });
 
 const reader = createConsoleReader();
-const memory = new UnconstrainedMemory();
-
+reader.write("Assistant 🤖 : ", "What location do you want to learn about?");
 for await (const { prompt } of reader) {
-  await memory.add(new UserMessage(prompt, { createdAt: new Date() }));
-
-  const { result } = await workflow.run(memory.messages).observe((emitter) => {
-    emitter.on("success", (data) => {
-      reader.write(`-> ${data.step}`, data.state?.finalAnswer ?? "-");
+  const { result } = await workflow
+    .run([
+      new AgentWorkflowInput("Provide a short history of the location.", prompt),
```

Comment:
I know you want to ensure 1:1 mapping with Python, but having just an interface in TS is more convenient.

Reply:
convert it into an interface